barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/goblin_translator_circuit_builder.cpp

```diff
@@ -99,54 +195,49 @@ GoblinTranslatorCircuitBuilder::AccumulationInput generate_witness_values(
-    auto [z_1_lo, z_1_hi] = split_wide_limb_into_2_limbs(z_1);
-    auto [z_2_lo, z_2_hi] = split_wide_limb_into_2_limbs(z_2);
+    auto [z_1_lo, z_1_hi] = split_wide_limb_into_2_limbs(z1);
+    auto [z_2_lo, z_2_hi] = split_wide_limb_into_2_limbs(z2);
 
     // The formula is `accumulator = accumulator⋅x + (op + v⋅p.x + v²⋅p.y + v³⋅z₁ + v⁴z₂)`. We need to compute the
     // remainder (new accumulator value)
 
-    Fq remainder = previous_accumulator * x + base_z_2 * v_quarted + base_z_1 * v_cubed + base_p_y * v_squared +
-                   base_p_x * v + base_op;
-    uint512_t quotient_by_modulus = uint_previous_accumulator * uint_x + uint_z_2 * uint_v_quarted +
-                                    uint_z_1 * uint_v_cubed + uint_p_y * uint_v_squared + uint_p_x * uint_v + uint_op -
+    Fq remainder = previous_accumulator * evaluation_input_x + base_z_2 * v_quarted + base_z_1 * v_cubed +
+                   base_p_y * v_squared + base_p_x * batching_challenge_v + base_op;
+
+    // We also need to compute the quotient
+    uint512_t quotient_by_modulus = uint_previous_accumulator * uint_x + uint_z2 * uint_v_quarted +
+                                    uint_z1 * uint_v_cubed + uint_p_y * uint_v_squared + uint_p_x * uint_v + uint_op -
                                     uint512_t(remainder);
 
     uint512_t quotient = quotient_by_modulus / uint512_t(Fq::modulus);
-    // constexpr uint512_t MAX_CONSTRAINED_SIZE = uint512_t(1) << 254;
-    // constexpr uint512_t MAX_Z_SIZE = uint512_t(1) << (NUM_LIMB_BITS * 2);
-    //  numeric::uint1024_t max_quotient =
-    //      (uint1024_t(MAX_CONSTRAINED_SIZE) * MAX_CONSTRAINED_SIZE * 3 + MAX_Z_SIZE * MAX_CONSTRAINED_SIZE * 2 + 4) /
-    //      modulus_u512;
-    //  info("Max quotient: ", max_quotient);
-    //  info("Max quotient range constraint: ", max_quotient.get_msb() + 1);
-
-    auto [remainder_0, remainder_1, remainder_2, remainder_3, remainder_prime] = base_element_to_bigfield(remainder);
-    std::array<Fr, 5> remainder_witnesses = { remainder_0, remainder_1, remainder_2, remainder_3, remainder_prime };
+
+    ASSERT(quotient_by_modulus == (quotient * uint512_t(Fq::modulus)));
+
+    // Compute quotient and remainder bigfield representation
+    auto [remainder_0, remainder_1, remainder_2, remainder_3] = base_element_to_bigfield(remainder);
+    std::array<Fr, NUM_BINARY_LIMBS> remainder_witnesses = { remainder_0, remainder_1, remainder_2, remainder_3 };
     auto [quotient_0, quotient_1, quotient_2, quotient_3, quotient_prime] = uint512_t_to_bigfield(quotient);
-    std::array<Fr, 5> quotient_witnesses = { quotient_0, quotient_1, quotient_2, quotient_3, quotient_prime };
+    std::array<Fr, NUM_BINARY_LIMBS> quotient_witnesses = { quotient_0, quotient_1, quotient_2, quotient_3 };
 
     // We will divide by shift_2 instantly in the relation itself, but first we need to compute the low part (0*0) and
     // the high part (0*1, 1*0) multiplied by a signle limb shift
```

Comment:
I think what these expressions are doing is not so complicated but they're fairly hard to read/check. I wonder if some combination of breaking them up into smaller components plus adding more comments would be helpful.

Reply:
I added a bit more in places, but I am not sure how to expand this